syntax = "proto3";

package demo_project;

option go_package = "github.com/wiqram/IG-Trading-Microservices/protos/project";

service ProjectSvc {
    rpc createProject (CreateProjectRequest) returns (ProjectResponse);
    rpc createTag (CreateTagRequest) returns (TagResponse);
    rpc getProject(GetProjectRequest) returns (ProjectResponse);
}

message CreateProjectRequest {
    string user_id = 1;
    string name = 2;
}

message CreateTagRequest {
    string user_id = 1;
    string name = 2;
    string project_id = 3;
}

message GetProjectRequest {
    string user_id = 1;
    string project_id = 2;
}

message ProjectResponse {
    string id = 1;
    string name = 2;
    repeated TagResponse tags = 3;
}

message TagResponse {
    string id = 1;
    string name = 2;
    string project_id = 3;
}